prowler/providers/github/lib/service/service.py

```diff
@@ -41,3 +43,53 @@ def __set_clients__(self, session):
                 f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
             )
         return clients
+
+    def _handle_github_api_error(
+        self, error, context: str, item_name: str, reraise_rate_limit: bool = False
+    ):
+        """Centralized GitHub API error handling"""
+        if isinstance(error, github.RateLimitExceededException):
+            logger.error(f"Rate limit exceeded while {context} '{item_name}': {error}")
+            if reraise_rate_limit:
+                raise
+        elif isinstance(error, github.GithubException):
+            if "404" in str(error):
+                logger.warning(f"'{item_name}' not found or not accessible")
+            elif "403" in str(error):
+                logger.warning(
+                    f"Access denied to '{item_name}' - insufficient permissions"
+                )
+            else:
+                logger.error(f"GitHub API error for '{item_name}': {error}")
+        else:
+            logger.error(
+                f"{error.__class__.__name__}[{error.__traceback__.tb_lineno}]: {error}"
+            )
+
+    def _get_organization_or_user(self, client, name: str):
```

Comment:
Is not this method to get the repositories of the org/user? The name is confusing.